hack/lib/start.sh

```diff
@@ -0,0 +1,496 @@
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - NODE_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Arguments:
+#  1 - API server version (ie. "v1.2.0")
+#  2 - Controllers version (ie. "v1.2.0")
+#  3 - Skip node start ("1" to skip node start)
+# Returns:
+#  - export OS_PID
+#  - export ETCD_PID
+#  - export API_SERVER_PID
+#  - export CONTROLLERS_PID
+#  - export NODE_PID
+function os::start::server() {
+	local api_server_version="${1:-}"
+	local controllers_version="${2:-}"
+	local skip_node="${3:-}"
+
+	if [[ -z "${api_server_version}" && -z "${controllers_version}" && -n "${skip_node}" ]]; then
+		os::start::master
+		return
+	fi
+
+
+	echo "[INFO] `openshift version`"
+	echo "[INFO] Server logs will be at:	${LOG_DIR}/openshift.log"
+	echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+	echo "[INFO] Volumes dir is:		${VOLUME_DIR}"
+	echo "[INFO] Config dir is:		${SERVER_CONFIG_DIR}"
+	echo "[INFO] Using images:		${USE_IMAGES}"
+	echo "[INFO] MasterIP is:		${MASTER_ADDR}"
+
+	mkdir -p ${LOG_DIR}
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
+
+	if [[ -z "${api_server_version}" && -z "${controllers_version}" && -z "${skip_node}" ]]; then
+		os::start::internal::start_all_in_one
+		return
+	fi
+
+	os::start::internal::start_etcd
+	os::start::internal::start_api_server "${api_server_version}"
+	os::start::internal::start_controllers "${controllers_version}"
+	if [[ -z "${SKIP_NODE}" ]]; then
+		os::start::internal::start_node
+	fi
+}
+readonly -f os::start::server
+
+# os::start::master starts the OpenShift master, exports the PID of the OpenShift master and waits until OpenShift master endpoints are available
+# It is advised to use this function after a successful run of 'os::start::configure_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+# Arguments:
+#  None
+# Returns:
+#  - export OS_PID
+function os::start::master() {
+	echo "[INFO] `openshift version`"
+	echo "[INFO] Server logs will be at:	${LOG_DIR}/openshift.log"
+	echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+	echo "[INFO] Config dir is:		${SERVER_CONFIG_DIR}"
+	echo "[INFO] Using images:		${USE_IMAGES}"
+	echo "[INFO] MasterIP is:		${MASTER_ADDR}"
+
+	mkdir -p ${LOG_DIR}
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
+	echo "[INFO] Starting OpenShift server"
+	local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
+	$(os::start::internal::openshift_executable) start master \
+		--config=${MASTER_CONFIG_DIR}/master-config.yaml \
+		--loglevel=4 --logspec='*importer=5' \
+	&>"${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
```

Comment:
it exports the PID of the container. The container is started just like any other process, and it's cleaned up by the regular cleanup trap.